kube/registry_credentials.go

```diff
@@ -0,0 +1,23 @@
+package kube
+
+import (
+	"github.com/SUSE/fissile/helm"
+)
+
+// MakeRegistryCredentials generates a template that contains Docker Registry credentials
+func MakeRegistryCredentials(createHelmChart bool) (helm.Node, error) {
+
+	value := ""
+	if createHelmChart {
+		value = "{{ printf \"{%s:{%s:%s,%s:%s,%s:%s}}\" (.Values.kube.registry.hostname | quote) (printf \"%s\" \"username\" | quote) (default \"\" .Values.kube.registry.username | quote) ( printf \"%s\" \"password\" | quote) (default \"\" .Values.kube.registry.password | quote) ( printf \"%s\" \"auth\" | quote) ( printf \"%s:%s\" .Values.kube.registry.username .Values.kube.registry.password | b64enc | quote ) | b64enc }}"
```

Comment:
This line is too long; I would like to see it broken up into smaller pieces that are then put together. Or find some other way to line wrap at significant points.

Use Go's backquotes to avoid having to escape double-quotes inside the string: `` `printf "user" ...` ``.

Also isn't `(printf "%s" "username" | quote)` the same as `("username" | quote)` (or `(printf "%q" "username")`)?

Reply:
What do you think of 86e4600?